src/jdk.jfr/share/classes/jdk/jfr/internal/util/Utils.java

```diff
@@ -243,67 +243,24 @@ public static void ensureInitialized(Class<? extends jdk.internal.event.Event> e
     }
 
     public static Object makePrimitiveArray(String typeName, List<Object> values) {
+        Class<?> componentType = makePrimitiveType(typeName);
+        if (componentType == null) {
+            return null;
+        }
         int length = values.size();
-        switch (typeName) {
-        case "int":
-            int[] ints = new int[length];
-            for (int i = 0; i < length; i++) {
-                ints[i] = (int) values.get(i);
-            }
-            return ints;
-        case "long":
-            long[] longs = new long[length];
-            for (int i = 0; i < length; i++) {
-                longs[i] = (long) values.get(i);
-            }
-            return longs;
-
-        case "float":
-            float[] floats = new float[length];
-            for (int i = 0; i < length; i++) {
-                floats[i] = (float) values.get(i);
-            }
-            return floats;
-
-        case "double":
-            double[] doubles = new double[length];
-            for (int i = 0; i < length; i++) {
-                doubles[i] = (double) values.get(i);
-            }
-            return doubles;
-
-        case "short":
-            short[] shorts = new short[length];
-            for (int i = 0; i < length; i++) {
-                shorts[i] = (short) values.get(i);
-            }
-            return shorts;
-        case "char":
-            char[] chars = new char[length];
-            for (int i = 0; i < length; i++) {
-                chars[i] = (char) values.get(i);
-            }
-            return chars;
-        case "byte":
-            byte[] bytes = new byte[length];
-            for (int i = 0; i < length; i++) {
-                bytes[i] = (byte) values.get(i);
-            }
-            return bytes;
-        case "boolean":
-            boolean[] booleans = new boolean[length];
-            for (int i = 0; i < length; i++) {
-                booleans[i] = (boolean) values.get(i);
-            }
-            return booleans;
-        case "java.lang.String":
-            String[] strings = new String[length];
-            for (int i = 0; i < length; i++) {
-                strings[i] = (String) values.get(i);
-            }
-            return strings;
+        Object array =  Array.newInstance(componentType, length);
+        for (int index = 0; index < length; index++) {
+            Array.set(array, index, values.get(index));
```

Comment:
Note that `Array.set` is slow (see [JDK-8051447](https://bugs.openjdk.org/browse/JDK-8051447)); yet this woudld be fine if this is not on hot code paths.

Reply:
It happens when a recording file is parsed, typically the Category annotation, which means about 150-200 invocation per file. Don't think it matters.

Reply:
Indeed, that sounds all good. If this becomes a performance issue, it should be fixed from java.lang.reflect instead.